contents/data/sql/SQL.md

```diff
@@ -0,0 +1,242 @@
+
+## Disadvantages
+
+SQL is not without its problems. In general, the biggest problem is with regards to the features of SQL.
+
+Although SQL databases use established <tooltip content="American National Standard Institutes">ANSI</tooltip> & <tooltip content="International Organization for Standardization">ISO</tooltip> standards, some distributions (PostgreSQL, for example) add proprietary extensions to the standard SQL to ensure customer lock-in.
+Thus, the available feature set varies according to the SQL distribution that you are using. This can make it confusing and frustrating to use SQL when switching across distributions.
+
+Apart from that, most **SQL** problems are not uniformly applicable to all of its distributions.
+
+For example, **MySQL** suffers from concurrency issues. Though it performs well with read operations, it *can* be problematic when there are many concurrent read-write operations. A symptom of this issue would be a sudden slowdown of a well-optimized query. However, **PostgreSQL** deals with concurrency well by being fully <tooltip content="Atomicity, Consistency, Isolation, Durability properties are satisfied">ACID</tooltip> compliant and implementing <tooltip content="transaction must be Isolated from other concurrent transactions running in the system">transactions isolation</tooltip>.
+
+---
+
+## How to get started with SQL?
+
+The first thing to do is to choose a SQL distribution. You can either just use an online SQL playground or install a dedicated application. My suggestion would be to use either MySQL or SQLite online.
+
+SQL has been around for a long time and hence, you can find really good books and resources to learn it well. However, these resources can be overwhelming. So, you can decide which parts you want to learn as you will likely not need to know everything to get started with using a SQL database.
+
+Here are some recommended steps that you can use to start learning SQL:
+
+- Understand what a database and [<tooltip content="DataBase Management System">DBMS</tooltip>]({{baseUrl}}/contents/data/databases/databases.html) are. More specifically, understand the basic concepts of [RDBMS](https://www.tutorialspoint.com/sql/sql-rdbms-concepts.htm). This is not _essential_, but will give you a better high-level understanding before diving into programming.
+
+- Khan Academy offers a very good [SQL course](https://www.khanacademy.org/computing/computer-programming/sql) for free. It has a mix of videos, text, and exercises that you can choose to do at your own pace. It is well organized and even provides summary notes.
+
+- W3Schools offers an interactive written tutorial for [SQL](https://www.w3schools.com/sql/default.asp).
+
+- You can also use a [SQL Quick Reference Guide](https://www.w3schools.com/sql/sql_ref_mysql.asp) to look up information quickly once you're familiar with the language.
+
+---
+
+## Where to go from here?
```

Comment:
Do you think choosing from the sql distribution first is the way to go?

I personally prefer to do it the other way round. Start with a web application and use a lightweight sql instance to store your data (e.g. sqlite?) Furthermore, many frameworks already provide database support by default (e.g. Django). 

This is okay as well!

Reply:
> Do you think choosing from the sql distribution first is the way to go?

I feel that in order to **learn** the language (not just use it), it's best to start from the basics. And, that means straight up writing queries to a database. 

> and use a lightweight sql instance to store your data (e.g. sqlite?)

Doesn't this involve implicitly choosing a distribution? 😛 

> Start with a web application and use a lightweight sql instance to store your data

Yes, but that requires you to develop a web application as well, right? And it will require you to have knowledge of the language as well (Python, in the case of Django). It won't be independent SQL, which is why I suggested starting out on MySQL/SQLite online instead.

Reply:
Yes fair enough! If you intend to learn pure SQL, this is the path.

I never actually started learning SQL on its own (and most people dont right?) -> its just something you pick up along the way when building bigger applications :)